addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,719 @@
+            # Because of that we wouldn't be able to differentiate a real issue (invoice on our side not matching theirs)
+            # With simply a status already up to date. So we need to check the status first to see if we need to update.
+            invoice_lookup, error_message = invoice._l10n_vn_edi_lookup_invoice()
+            if error_message:
+                raise UserError(error_message)
+
+            if 'result' in invoice_lookup:
+                invoice_data = invoice_lookup['result'][0]
+                if invoice_data['status'] == 'Chưa thanh toán':  # Vietnamese for 'unpaid'
+                    sinvoice_status = 'unpaid'
+                else:
+                    sinvoice_status = 'paid'
+
+            if invoice.payment_state in {'in_payment', 'paid'} and sinvoice_status == 'unpaid':
+                # Mark the invoice as paid
+                endpoint = 'InvoiceAPI/InvoiceWS/updatePaymentStatus'
+                params = {
+                    'supplierTaxCode': invoice.company_id.vat,
+                    'templateCode': invoice.l10n_vn_edi_invoice_symbol.invoice_template_id.name,
+                    'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                    'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+                }
+            elif invoice.payment_state not in {'in_payment', 'paid'} and sinvoice_status == 'paid':
+                # Mark the invoice as not paid
+                endpoint = 'InvoiceAPI/InvoiceWS/cancelPaymentStatus'
+                params = {
+                    'supplierTaxCode': invoice.company_id.vat,
+                    'invoiceNo': invoice.l10n_vn_edi_invoice_number,
+                    'strIssueDate': invoice._l10n_vn_edi_format_date(invoice.l10n_vn_edi_issue_date),
+                }
+
+            if endpoint and params:
+                _request_response, error_message = invoice._l10n_vn_edi_send_request(
+                    method='POST',
+                    endpoint=endpoint,
+                    params=params,
+                )
+
+                if error_message:
+                    raise UserError(error_message)
+
+                # Revert back to the sent state as the status is up-to-date.
+                invoice.l10n_vn_edi_state = 'sent'
+
+    def _need_cancel_request(self):
+        # EXTEND 'account'
+        return super()._need_cancel_request() or self.l10n_vn_edi_invoice_number
+
+    def _post(self, soft=True):
+        # EXTEND 'account'
+        posted = super()._post(soft)
+
+        # Ensure to tag the move as 'Ready to send' upon posting if it makes sense.
+        posted.filtered(
+            lambda invoice:
+                invoice.company_id.account_fiscal_country_id.code == 'VN'
+                and invoice.is_sale_document()
+                and not invoice.l10n_vn_edi_invoice_number
+                and not invoice.l10n_vn_edi_state
+        ).l10n_vn_edi_state = 'ready_to_send'
```

Comment:
I don't think it should.
It's not just about the technical part, "ready_to_send" allows users to filter in the list view all invoices that are ready to be sent to the EDI/it will enable to add decorations on the tree view etc.

We must keep it for UX purposes or replace it with a compute that tags invoices as 'SInvoice' based on the same conditions. (and then we can use "no state" + "is Sinvoice" for the same purpose).

This state doesn't remove the checkbox or any usability of the send & print wizard.